File: backend/shared/models/conversation.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional
from uuid import UUID

class ConversationBase(BaseModel):
    title: str

class ConversationCreate(ConversationBase):
    pass

class ConversationResponse(ConversationBase):
    id: UUID
    user_id: UUID
    preview_text: Optional[str] = None
    message_count: int = 0
    is_pinned: bool = False
    last_accessed_at: datetime
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

class ConversationUpdate(BaseModel):
    title: Optional[str] = None
    is_pinned: Optional[bool] = None

File: lib/core/models/conversation.dart
/// 对话模型
class Conversation {
  final String id;
  final String title;
  final String? previewText;
  final int messageCount;
  final bool isPinned;
  final DateTime lastAccessedAt;
  final DateTime createdAt;
  final DateTime? updatedAt;

  Conversation({
    required this.id,
    required this.title,
    this.previewText,
    this.messageCount = 0,
    this.isPinned = false,
    required this.lastAccessedAt,
    required this.createdAt,
    this.updatedAt,
  });

  factory Conversation.fromJson(Map<String, dynamic> json) {
    return Conversation(
      id: json['id']?.toString() ?? '',
      title: json['title'] ?? '',
      previewText: json['previewText'],
      messageCount: json['messageCount'] ?? 0,
      isPinned: json['isPinned'] ?? false,
      lastAccessedAt: json['lastAccessedAt'] != null
          ? DateTime.parse(json['lastAccessedAt'])
          : DateTime.now(),
      createdAt: json['createdAt'] != null
          ? DateTime.parse(json['createdAt'])
          : DateTime.now(),
      updatedAt: json['updatedAt'] != null
          ? DateTime.parse(json['updatedAt'])
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'previewText': previewText,
      'messageCount': messageCount,
      'isPinned': isPinned,
      'lastAccessedAt': lastAccessedAt.toIso8601String(),
      'createdAt': createdAt.toIso8601String(),
      'updatedAt': updatedAt?.toIso8601String(),
    };
  }
}
